enhancements/proposals/1304-external-seeds/README.md

```diff
@@ -0,0 +1,156 @@
+
+## Design Details
+
+### ScyllaCluster API
+
+The following extension to the existing ScyllaCluster CR API is proposed:
+```go
+// ScyllaClusterSpec defines the desired state of Cluster.
+type ScyllaClusterSpec struct {
+    ...
+	
+	// externalSeeds specifies the external seeds to propagate to ScyllaDB binary on startup as "seeds".
+	// See https://opensource.docs.scylladb.com/stable/kb/seed-nodes.html.
+	// For the first running node in a datacenter, the external seeds are the only seeds provided.
+	// For the remaining nodes, one other node from their datacenter is appended to the external seeds.
+	// +optional
+	ExternalSeeds []string `json:"externalSeeds,omitempty"`
+}
+```
+
+The provided `externalSeeds` are passed to the sidecar through a new `--external-seeds` flag. StatefulSet reconciliation in ScyllaDB operator is modified accordingly.
+The sidecar's strategy for choosing seeds to provide to ScyllaDB processes is modified to follow the below pseudocode:
+```go
+func chooseSeeds(externalSeeds []string, seedFromLocalDC string) []string {
+  if is first node in DC {
+    if externalSeeds not empty {
+      return externalSeeds
+    }
+    return seedFromLocalDC
+  }
+
+  return append(externalSeeds, seedFromLocalDC)
+}
+```
+
+It is important that the external seeds, and only the external seeds, are provided as seeds for the first running node in a datacenter, if they are at all provided by the end user. If external seeds were only appended to the node's own IP, it could result in creating split clusters in case of lack of connectivity between the datacenters.
+For the other nodes, a seed from their own DC is appended to the external seeds for increased resiliency.
+
+The above changes allow for configuring multi datacenter ScyllaDB clusters by manually providing external seeds.
+The below examples assume the two datacenters are to be deployed in one Kubernetes cluster.
+
+### Creating a multi datacenter ScyllaDB cluster
+To create a multi datacenter ScyllaDB cluster, an end user would first deploy a ScyllaCluster. For the purpose of this document, let's assume name `simple-cluster`, namespace `scylla1`, datacenter `us-east-1` and a single rack `us-east-1a`.
+To create the second datacenter, the end user would deploy a second ScyllaCluster with the same name, in a different namespace, with a distinct datacenter and rack names, and `.spec.externalSeeds` set to a list of IP address or domain names under which one or more nodes of the first datacenter are reachable from the second datacenter. 
+ScyllaDB operator will propagate the declared `externalSeeds` through a sidecar flag, and the sidecar will take it into account when configuring the seeds provided to ScyllaDB nodes of the second datacenter. The two datacenters will then form a single multi datacenter cluster.
+
+### Adding a new datacenter to an existing ScyllaDB cluster
+To add a new datacenter to an existing ScyllaCluster, the same procedure as above is followed.
```

Comment:
Can we explicitly mention this working for connecting 2 existing and rolled out ScyllaClusters?

Reply:
Connecting two bootstrapped DCs isn't guaranteed with just a rollout. Since both DCs are already bootstrapped, patching one of them with external seeds will only result in forming a multi DC cluster if at least one of the nodes falls back to using initial contact points, i.e. only if it is not able to talk with any of the other nodes from its own DC, as described in `Current role of seeds in ScyllaDB` section of this proposal. All of that despite the fact that patching an existing DC with external seeds results in following the procedure described in [this page of ScyllaDB docs]( https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/cluster-management/add-dc-to-existing-dc.html). It's probably outdated, since I don't see how just restarting the nodes of a bootstrapped cluster one by one would always result in forming a multi DC cluster. I don't have any ideas for covering this case easily, but I'm actually wondering whether this should even be expected to be covered?

Reply:
ok, let's list it as a non-goal then and be aware that the the DC can be join only when being created.